lib/pbench/server/database/models/datasets.py

```diff
@@ -25,51 +25,47 @@ class DatasetError(Exception):
     It is never raised directly, but may be used in "except" clauses.
     """
 
+    pass
+
 
 class DatasetBadName(DatasetError):
     """Specified filename does not follow Pbench tarball naming rules."""
 
     def __init__(self, name: Path):
+        super().__init__(
+            f"File name {name!r} does not end in {Dataset.TARBALL_SUFFIX!r}"
+        )
         self.name: str = str(name)
```

Comment:
I like having the attributes, because they're easier to reference than embedded text.

In one sense it doesn't make much difference since we don't propagate exceptions outside our code and most of these only get consumed by unit tests. Still, it's a nice convention and something of a habit.